source/common/stats/stat_merger.cc

```diff
@@ -136,6 +136,23 @@ void StatMerger::mergeGauges(const Protobuf::Map<std::string, uint64_t>& gauges,
   }
 }
 
+void StatMerger::retainParentGaugeValue(Stats::StatName gauge_name) {
+  parent_gauge_values_.erase(gauge_name);
+}
+
+void StatMerger::removeParentContributionToGauges() {
+  for (auto& iter : parent_gauge_values_) {
+    Gauge& gauge = temp_scope_->gaugeFromStatName(iter.first, Gauge::ImportMode::Accumulate);
+    if (gauge.importMode() == Gauge::ImportMode::NeverImport) {
```

Comment:
Can you add some comments here? I assume this is the case where the gauge type has changed in the child?